util/dvsim/FlowCfg.py

```diff
@@ -411,36 +411,52 @@ def gen_results(self, results):
 
         '''
         for item in self.cfgs:
+            json_str = (item._gen_json_results(results)
+                        if hasattr(item, '_gen_json_results')
+                        else None)
             result = item._gen_results(results)
             log.info("[results]: [%s]:\n%s\n", item.name, result)
             log.info("[scratch_path]: [%s] [%s]", item.name, item.scratch_path)
-            item.write_results_html(self.results_html_name, item.results_md)
+            item.write_results(self.results_html_name, item.results_md,
```

Comment:
I would rename this field to `results_filename`, and change the caller to don't add the file extention. So the function `write_results` would choose the file extention based on the content.

Reply:
I had considered that at first, but a problem I found is that `results_html_name` gets its value from the hjson config file, e.g.:
https://github.com/lowRISC/opentitan/blob/90e0fa6b338f11c5db894a667615daa32e0a0eb7/hw/data/common_project_cfg.hjson#L29

So we could:
a) Change the field in the hjson to `results_base_name` and its value in `common_project_cfg.hjson` to `"report"`. This would be the cleanest but also the most invasive change.
b) Keep the field in the hjson but derive a `results_base_name` after deserializing the hjson. This would move the string manipulation further up the call chain.
c) Use the current solution, where the suffix is replaced when/if necessary locally in the code that needs it.

I went for (c) because I found it the simplest. What do you prefer?
c)